src/content/app/genome-browser/components/browser-sidebar-modal/modal-views/navigate-modal/NavigateLocationModal.tsx

```diff
@@ -0,0 +1,344 @@
+    setLocationEndInput(value);
+    setLocationEndErrorMessage(null);
+  };
+
+  const onLocationInputChange = (event: FormEvent<HTMLInputElement>) => {
+    const value = event.currentTarget.value;
+    setLocationInput(value);
+    setLocationErrorMessage(null);
+  };
+
+  const handleSubmit = () => {
+    if (activeGenomeId) {
+      setLocationErrorMessage(null);
+
+      const newLocation = coordInputsActive
+        ? `${regionNameInput}:${locationStartInput}-${locationEndInput}`
+        : locationInput;
+
+      validateRegion({
+        regionInput: newLocation,
+        genomeId: activeGenomeId,
+        onSuccess: onValidationSuccess,
+        onError: onValidationError
+      });
+    }
+  };
+
+  const getKaryotypeOptions = () =>
+    genomeKaryotype.map(({ name }) => ({
+      value: name,
+      label: name,
+      isSelected: regionNameInput === name
+    }));
+
+  const updateRegionNameInput = (event: FormEvent<HTMLSelectElement>) => {
+    const value = event.currentTarget.value;
+
+    setCoordInputsActive(true);
+    setLocationInputActive(false);
+
+    const selectedKaryotypeItems = genomeKaryotype.find(
+      ({ name }) => name === value
+    );
+
+    if (selectedKaryotypeItems) {
+      setRegionNameInput(value);
+    }
+  };
+
+  const updateErrorMessages = (
+    locationStartError: string | null,
+    locationEndError: string | null,
+    locationError: string | null
+  ) => {
+    if (coordInputsActive) {
+      setLocationStartErrorMessage(locationStartError);
+      setLocationEndErrorMessage(locationEndError);
+    } else {
+      setLocationErrorMessage(
+        locationStartError || locationEndError || locationError
+      );
+    }
+  };
+
+  const onValidationError = (errorMessages: RegionValidationErrors) => {
+    const {
+      startError = null,
+      endError = null,
+      regionError = null
+    } = errorMessages;
+    updateErrorMessages(startError, endError, regionError);
+  };
+
+  const resetForm = () => {
+    setRegionNameInput('');
+    setLocationStartInput('');
+    setLocationEndInput('');
+    setLocationInput('');
+
+    setCoordInputsActive(false);
+    setLocationInputActive(false);
+
+    updateErrorMessages(null, null, null);
+  };
+
+  const onValidationSuccess = () => {
+    resetForm();
+
+    let newChrLocation: ChrLocation;
+    let newChrLocationStr: string;
+
+    if (coordInputsActive) {
+      newChrLocation = [
+        regionNameInput,
+        getNumberWithoutCommas(locationStartInput),
+        getNumberWithoutCommas(locationEndInput)
+      ];
+      newChrLocationStr = `${regionNameInput}:${locationStartInput}-${locationEndInput}`;
+    } else {
+      newChrLocation = getChrLocationFromStr(locationInput);
+      newChrLocationStr = locationInput;
+    }
+
+    changeBrowserLocation({
+      genomeId: activeGenomeId as string,
+      chrLocation: newChrLocation,
+      focus: {
+        type: 'location',
+        id: `location:${newChrLocationStr}`
+      }
+    });
```

Comment:
I tried using this to change the focus object. However, this doesn't work. Should it be done different?

Reply:
- create a new genome browser url like [here](https://github.com/Ensembl/ensembl-client/pull/906/files#diff-a31338b9080b764e86eb24c99ee592ef35fa20e46932d3f212e9ea5691abd214R489-R492)
- navigate to that url

Reply:
✅